Modify rule S2014: Migration to new educational format (#1896)
SonarSource/rspec: rules/S2014/php/rule.adoc

== Why is this an issue?

> In PHP, `$this` is a keyword used inside a class, which refers to the current class instance.
> You have access to class properties and methods through the use of this keyword.

Reviewer: ```suggestion
To reference the current class instance the keyword $this can be used.
Through the use of this keyword you have access to class properties and methods.
```

> In PHP, `$this` is a keyword used inside a class, which refers to the current class instance.
> You have access to class properties and methods through the use of this keyword.

Reviewer: ```suggestion
To reference the current class instance the keyword `$this` can be used.
Through the use of this keyword, you have access to class properties and methods.
```

> Static methods can be accessed without instantiating the class, so ``++$this++`` is not available for them.
> Using ``++$this++`` in a static context will result in a runtime error.

Reviewer: ```suggestion
Static methods can be accessed without instantiating the class, so `$this` is not available for them.
Using `$this` in a static context will result in a runtime error.
```
So far I know the `++` are left overs from a migration. We used to have this data stored in Jira before RSPEC was moved to GitHub.
